### PR TITLE
Ant normalisation

### PR DESCRIPTION
**Pull request by surajit4444mondal**
Changes made to incorporate antenna normalisation. Following changes made:

1. Flags present in caltables will be respected.
2. Additional antennas can be passed via flag_ants list for flagging

Does not support additional weighting of antennas. If using additional weighting, additional calibration step is necessary.

### CODE REVIEW

File: mnc/xengine_beamformer_control.py [hunk 1]
```diff
@@ -21,6 +21,9 @@
 from astropy.constants import c as speedOfLight
 speedOfLight = speedOfLight.to('m/ns').value
 
+print (speedOfLight)
```
Comment (caseyjlaw) [inline mnc/xengine_beamformer_control.py:24]: Looks like a debugging statement. Could we remove it?

File: mnc/xengine_beamformer_control.py [hunk 2]
```diff
@@ -287,11 +290,27 @@ def set_beam_calibration(self, caltable, verbose=True):
         if len(subband_pipelines) != NPIPELINE_SUBBAND:
             logger.warning(f"Found {len(subband_pipelines)} pipelines associated with these data instead of the expected {NPIPELINE_SUBBAND}")
             
+            
+        num_chan=len(subband_pipelines)*NCHAN_PIPELINE
+        norm_factors=numpy.ones((num_chan,NPOL),dtype=float)
+        flagged_ant=numpy.array(flag_ants)
+       
+        print (flgdata.shape)
```
Comment (caseyjlaw) [inline mnc/xengine_beamformer_control.py:298]: Can we remove that or use the logger?

**Review by caseyjlaw — COMMENTED** [review]
Looks great. Minor clean up requested.

File: mnc/xengine_beamformer_control.py (hunk 1 above)
Comment (surajit4444mondal) [reply mnc/xengine_beamformer_control.py]: Done

**Comment by surajit4444mondal** [comment]
Done.

**Review by caseyjlaw — APPROVED** [review]

**Pull request merged**